src/client/workspaces.ts

```
@@ -496,3 +496,110 @@ export async function restore(payload: Workspace): Promise<Workspace> {
+        return createChild(win);
+    } else {
+        // only position if the window exists
+        return positionWindow(win);
+    }
+}
+
+/**
+ * Given a workspaceWindow object for a non-open window, this will create the window, position it,
+ * and update its visibility, state and frame to fit with the restoring workspace.
+ *
+ * Called by the defaultRestoreHandler for any non-running windows in the workspace
+ */
+export async function createChild(workspaceWindow: WorkspaceWindow): Promise<void> {
+    const {bounds, frame, isShowing, name, state, url} = workspaceWindow;
+    await fin.Window.create({
+        url,
+        autoShow: isShowing,
+        defaultHeight: bounds.height,
+        defaultWidth: bounds.width,
+        defaultLeft: bounds.left,
+        defaultTop: bounds.top,
+        frame,
+        state,
+        name
+    });
+}
+
+/**
+ * Given a workspaceWindow object for a currently open window, this will position and
+ * update the visibility, state and frame of that window to fit with the restoring
+ * workspace.
+ *
+ * Called by the defaultRestoreHandler for any currently running windows in the workspace.
+ */
+export async function positionWindow(workspaceWindow: WorkspaceWindow): Promise<void> {
+    const {bounds, frame, isShowing, state, isTabbed} = workspaceWindow;
+    try {
+        const ofWin = fin.Window.wrapSync(workspaceWindow);
+        await ofWin.setBounds(bounds);
+
+        if (isTabbed) {
+            return;
+        }
+
+        if (!isShowing) {
+            await ofWin.hide();
+            return;
+        }
+
+        if (state === 'normal') {
+            await ofWin.restore();
+        } else if (state === 'minimized') {
+            await ofWin.minimize();
+        } else if (state === 'maximized') {
+            await ofWin.maximize();
+        }
+
+        await ofWin.updateOptions({frame});
+    } catch (e) {
+        console.error('position window error: ' + e);
```

Need to think about how to handle errors in these handlers. This one suppresses errors (with a console log), but `createChild` will throw/reject.

Failure cases should also be called-out in the docs, IMO. Other than that, docs look good.

Let's discuss as a follow-up